### ISSUE TITLE
SCSS not precompiling 

### ISSUE THREAD

**Issue by vladiim** [hanami/hanami#615]
* `apps/cms/application.rb` has `require 'sass'`
* `apps/cms/assets/stylesheets/application.css.scss` has some basic css
* `apps/cms/templates/application.html.erb` includes the file with `<%= stylesheet 'application' %>`
* When I run `hanami server` an `application.css` file is generated with no contents
* When I run `hanami assets precompile` I get the error below, note `compile true` has been added to `apps/cms/application.rb`

**Key error from `hanami assets precompile`**

    /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/find.rb:43:in `block in find': undefined method `encoding' for #<Pathname:/Users/vlad.mehakovic/Code/dynomail/apps/cms/assets> (NoMethodError)

**Full error from `hanami assets precompile`**

```
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/json/version.rb:3: warning: already initialized constant JSON::VERSION
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/json-1.8.3/lib/json/version.rb:3: warning: previous definition of VERSION was here
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/json/version.rb:4: warning: already initialized constant JSON::VERSION_ARRAY
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/json-1.8.3/lib/json/version.rb:4: warning: previous definition of VERSION_ARRAY was here
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/json/version.rb:5: warning: already initialized constant JSON::VERSION_MAJOR
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/json-1.8.3/lib/json/version.rb:5: warning: previous definition of VERSION_MAJOR was here
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/json/version.rb:6: warning: already initialized constant JSON::VERSION_MINOR
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/json-1.8.3/lib/json/version.rb:6: warning: previous definition of VERSION_MINOR was here
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/json/version.rb:7: warning: already initialized constant JSON::VERSION_BUILD
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/json-1.8.3/lib/json/version.rb:7: warning: previous definition of VERSION_BUILD was here
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/json/common.rb:99: warning: already initialized constant JSON::NaN
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/json-1.8.3/lib/json/common.rb:99: warning: previous definition of NaN was here
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/json/common.rb:101: warning: already initialized constant JSON::Infinity
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/json-1.8.3/lib/json/common.rb:101: warning: previous definition of Infinity was here
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/json/common.rb:103: warning: already initialized constant JSON::MinusInfinity
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/json-1.8.3/lib/json/common.rb:103: warning: previous definition of MinusInfinity was here
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/json/common.rb:128: warning: already initialized constant JSON::UnparserError
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/json-1.8.3/lib/json/common.rb:128: warning: previous definition of UnparserError was here
/Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/find.rb:43:in `block in find': undefined method `encoding' for #<Pathname:/Users/vlad.mehakovic/Code/dynomail/apps/cms/assets> (NoMethodError)
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/find.rb:42:in `each'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/2.1.0/find.rb:42:in `find'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/compiler.rb:213:in `block in sass_load_paths'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-utils-0.7.2/lib/hanami/utils/load_paths.rb:65:in `block in each'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-utils-0.7.2/lib/hanami/utils/load_paths.rb:64:in `each'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-utils-0.7.2/lib/hanami/utils/load_paths.rb:64:in `each'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/compiler.rb:212:in `sass_load_paths'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/compiler.rb:175:in `block in compile!'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/compiler.rb:197:in `block in write'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/compiler.rb:196:in `open'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/compiler.rb:196:in `open'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/compiler.rb:196:in `write'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/compiler.rb:175:in `compile!'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/compiler.rb:100:in `compile'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/compiler.rb:59:in `compile'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/precompiler.rb:59:in `block (2 levels) in precompile'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/precompiler.rb:58:in `each'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/precompiler.rb:58:in `block in precompile'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/precompiler.rb:49:in `each'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/precompiler.rb:49:in `precompile'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets/precompiler.rb:34:in `run'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-assets-0.2.1/lib/hanami/assets.rb:52:in `deploy'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-0.7.3/lib/hanami/commands/assets/precompile.rb:30:in `precompile'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-0.7.3/lib/hanami/commands/assets/precompile.rb:14:in `start'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-0.7.3/lib/hanami/cli_sub_commands/assets.rb:17:in `precompile'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:115:in `invoke'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor.rb:235:in `block in subcommand'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/hanami-0.7.3/bin/hanami:5:in `<top (required)>'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/bin/hanami:22:in `load'
	from /Users/vlad.mehakovic/.rbenv/versions/2.1.2/bin/hanami:22:in `<main>'
```

**Comment by jodosha** [comment]
@vladiim Hi, I'm sorry about your problem. I personally improved assets serving and precompilation in master, which is different from Hanami 0.7.3 (the version you're using). These improvements will be released at the end of the week with Hanami 0.8.0.

Would you be so kind to check if master solves this problem for you? If not, I'll make sure to fix it before the release. Please follow these few steps: https://github.com/hanami/hanami#how-to-use-hanami-head

Thank you very much.

**Comment by vladiim** [comment]
@jodosha thanks - no worries.

I tried to start a new app on 0.8.0 with `hanami new --hanami-head=true appname` but when I ran `bundle install` I got an error:

```
NoMethodError: undefined method `spec' for nil:NilClass
An error occurred while installing hanami-utils (0.8.0), and Bundler cannot continue.
```

**Comment by jodosha** [comment]
@vladiim Did you ran that command **from the cloned repository of `hanami/hanami`**?

**Comment by vladiim** [comment]
@jodosha I wasn't, working now 👍

**Comment by jodosha** [comment]
Fixed by #612

**Comment by jodosha** [comment]
@vladiim Thanks for checking. I've closed this. BTW by the end of this week, we'll release Hanami v0.8, which will include this fix.
